ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -579,19 +589,25 @@ applyPatchCmmParseNoImplicitPrelude ghcFlavor =
 -- function 'appyPatchHadrianStackYaml' guarantees it is available
 -- if its needed.
 
--- Patch Hadrian's Cabal.
+-- | Patch Hadrian's Cabal.
 applyPatchHadrianStackYaml :: GhcFlavor -> IO ()
 applyPatchHadrianStackYaml ghcFlavor = do
-   appendFile "hadrian/stack.yaml" $ unlines ["ghc-options:","  \"$everything\": -O0 -j"]
-   case ghcFlavor of
-     GhcMaster ->
-       appendFile "hadrian/stack.yaml" $
-       unlines [
-           "extra-deps:"
-         , "  - happy-1.20.0" -- See https://gitlab.haskell.org/ghc/ghc/-/issues/18726.
-         , "  - exceptions-0.10.4" -- See [Note : GHC now depends on exceptions package]
-         ]
-     _ -> pure ()
+  let hadrianStackYaml = "hadrian/stack.yaml"
+  bytes <- B.readFile hadrianStackYaml
+  config <- Y.decodeThrow bytes
+  -- See [Note : GHC now depends on exceptions package]
+  let deps = [x | ghcFlavor == GhcMaster, x <- [toJSON (T.pack "exceptions-0.10.4")]] ++
```

Comment:
`Y.Value` has an `IsString` instance and we don't need the binding:
```suggestion
  let deps = ["exceptions-0.10.4" | ghcFlavor == GhcMaster] ++
```

Reply:
Ok